src/Disks/IO/CachedOnDiskWriteBufferFromFile.cpp

```diff
@@ -88,39 +178,25 @@ bool FileSegmentRangeWriter::write(const char * data, size_t size, size_t offset
     if (downloader != FileSegment::getCallerId())
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Failed to set a downloader. ({})", file_segment->getInfoForLog());
 
-    SCOPE_EXIT({
-        if (file_segment->isDownloader())
-            file_segment->completePartAndResetDownloader();
-    });
```

Comment:
`completePartAndResetDownloader` is called outside of this function (because it reserves space and we also need to write something or advance offset in case we need only to reserve). I'll check that it's called (by adding `SCOPE_EXIT` as you suggested in comment above https://github.com/ClickHouse/ClickHouse/pull/43972#discussion_r1042037456